lib/matplotlib/colors.py

```diff
@@ -286,6 +286,8 @@ def to_rgba(c, alpha=None):
         Tuple of floats ``(r, g, b, a)``, where each channel (red, green, blue,
         alpha) can assume values between 0 and 1.
     """
+    if isinstance(c, tuple) and len(c) == 2:
+        c, alpha = c[0], c[1]
```

Comment:
I think `to_rgba` does not have to do any special-casing. It simply passes this on to `to_rgba_no_colorcycle()`, which indeed must handle (color, alpha) tuples on its own. So move this conversion into `to_rgba_no_colorcycle()`.

Note that, neither the `_is_nth_color` logic nor `_colors_full_map.cache` do something unexpected with (color, alpha) tuples.

Additionally, we want to keep 

> If *alpha* is given, force the alpha value of the returned RGBA tuple
to *alpha*.

You cannot reassign alpha unconditionally. I think you need to
```
if alpha is None:
    c, alpha = c
else:
    c = c[0]  # discard alpha from tuple
```